### PR TITLE
Memory Leak Test for php unit tests

### PR DESCRIPTION
**Pull request by jerrycmh**
removed version control markers

This is reintroducing work originally merged in  https://github.com/grpc/grpc/pull/17728 (later reverted in https://github.com/grpc/grpc/pull/17852)

### CODE REVIEW

**Comment by stanley-cheung** [comment]
#17728

**Comment by stanley-cheung** [comment]
Building PHP is currently broken because of a security breach at PEAR: https://twitter.com/pear/status/1088155252644372482. Rebuilding our docker images currently failed.  According to http://pear.php.net/ there is no ETA of a fix yet.

**Comment by stanley-cheung** [comment]
We finally reached a state where all tests passed. The next set of potentials things we can do to make this better are:

 - Add (back) more tests to `MemoryLeakTest.php`
   - Within this, we don't need a lot of the tests found under the `unit_tests` directory.
   - In particular, we don't need some of those "constructor takes 2 parameter, let's pass it 0 parameter to see if it errors out" type of tests. We are only interested in exercising as much code as possible to check for memory leaks.
   - But we still should cover all the `Grpc\*` classes introduced in the `grpc` extension
   - In particular, there may be memory leaks associated with the `Server->requestCall` function
 - Add the tests to the PHP5 base docker image
   - There seems to be still come error if we run this the php5 base docker images.
   - But on the other hand, PHP5 is being phased out so this is less priority

We have pushed the relevant `tools/dockerfile/test` docker images to docker hub `grpctesting` org.

File: templates/tools/dockerfile/php_valgrind.include [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#=================	
+# PHP Test dependencies	
+
+ # Install dependencies
```
Comment (jtattermusch) [inline templates/tools/dockerfile/php_valgrind.include]: nit: the "install dependencies" comment is not very useful :-)

File: templates/tools/dockerfile/php_valgrind.include [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#=================	
+# PHP Test dependencies	
+
+ # Install dependencies	
+
+ RUN apt-get update && apt-get install -y ${'\\'}	
+    valgrind
```
Comment (jtattermusch) [inline templates/tools/dockerfile/php_valgrind.include]: nit: missing EOL.

**Comment by stanley-cheung** [comment]
Known test failure from Python: #17927

**Pull request merged**